src/ripple/protocol/impl/LedgerFormats.cpp

```diff
@@ -56,6 +56,7 @@ LedgerFormats::LedgerFormats()
             {sfMintedNFTokens,       soeDEFAULT},
             {sfBurnedNFTokens,       soeDEFAULT},
             {sfFirstNFTokenSequence, soeOPTIONAL},
+            {sfAMMID,                soeOPTIONAL},
```

Comment:
I'm not going to hate you for this and it's not a big deal to change. But I don't quite like the account type inference as `AMMID` field side effect. It might also be a problem for the XRPL explorer apps.

Reply:
Just a thought (feel free to ignore) - It may be possible to avoid changing the external API by returning a synthetic boolean field based on the presence of AMMID

Reply:
> It might also be a problem for the XRPL explorer apps.

Yeah, those are some of the people I expect to hate me. :laughing: 

But kidding aside, it's not like we're repurposing a field to hold the AMMID. This is a new dedicated field, exclusively (at least for now) for AMMs. If it's populated, then it's an AMM. I don't think it's an inference, I think it's a clear flag.

> It may be possible to avoid changing the external API by returning a synthetic boolean field based on the presence of AMMID

If you're talking about the recently-added `account_flags` returned in `account_info`, I notice that it doesn't have support for the `lsfAMM` flag anyway. I think it would be perfectly sensible to add a synthetic "AMM" flag there.

However, anything else that returns the raw `AccountRoot` object would be a little harder, but it'll have the AMMID field, so explorers will have to check for that. The other thing that comes to mind is transaction metadata. I'm pretty sure that'll have the AMMID in the metadata regardless, though. Need to confirm.

Reply:
> Need to confirm.

Confirmed:
```
      {
         "ModifiedNode" : {
            "FinalFields" : {
               "AMMID" : "27ED2C339D335D3CA09BB6C85FF19685A65C00A105D68242383EDB3E06942757",
               "Account" : "rnV55X6wNHXQNostoLPQYZZPzBmDnSb14K",
               "Balance" : "11000000000",
               "Flags" : 1099956224,
               "OwnerCount" : 1,
               "Sequence" : 6
            },
            "LedgerEntryType" : "AccountRoot",
            "LedgerIndex" : "DF7F696214208B2C7FB92F28DE6FEE45921D159EF11D7D924294A18CE4A827E7",
            "PreviousFields" : {
               "Balance" : "10000000000"
            },
            "PreviousTxnID" : "7297888514F54699417843AEB220730B00D8EA2B2B06C2EF218A23285922EE7A",
            "PreviousTxnLgrSeq" : 6
         }
      },
```
This would be good to check in a unit test somewhere.